src/Curl/Curl.php

```diff
@@ -543,8 +543,10 @@ public function setDigestAuthentication($username, $password = '')
      */
     public function setCookie($key, $value)
     {
-        $this->cookies[$key] = $value;
-        $this->setOpt(CURLOPT_COOKIE, str_replace(' ', '%20', urldecode(http_build_query($this->cookies, '', '; '))));
+        $this->cookies[$key] = $value;        
+        $this->setOpt(CURLOPT_COOKIE, implode('; ', array_map(function($name) { 
+            return $name . '=' . str_replace(' ', '%20', $this->cookies[$name]); 
+        }, array_keys($this->cookies)))); 
```

Comment:
Can you remove the trailing whitespace from these lines? Thanks

![whitespace](https://cloud.githubusercontent.com/assets/1083146/10761739/f54479be-7c7f-11e5-8985-1437cdca1e07.png)